src/qutip_qip/circuit/text_renderer.py

```diff
@@ -0,0 +1,506 @@
+"""
+Module for rendering a quantum circuit in text format.
+"""
+
+from typing import List
+
+from ..operations import Gate, Measurement
+from . import QubitCircuit
+
+__all__ = [
+    "TextRenderer",
+]
+
+
+class TextRenderer:
+    """
+    A class to render a quantum circuit in text format.
+    """
+
+    def __init__(self, qc: QubitCircuit):
+
+        self._qc = qc
+        self._qwires = qc.N
```

Comment:
A design question, technically speaking, the Renderer class should be independent of the quantum circuit provided.

Therefore, `qc` should better be an input argument for some `canvas_plot` for `draw` methods. Not as part of the initialization of the renderer class.